kotlin-asyncapi-context/src/main/kotlin/org/openfolder/kotlinasyncapi/context/annotation/AnnotationProvider.kt

```diff
@@ -70,16 +72,24 @@ class AnnotationProvider(
                 listOfNotNull(
                     clazz.findAnnotation<Message>()?.let { clazz to it },
                     clazz.findAnnotation<Schema>()?.let { clazz to it },
-                    clazz.findAnnotation<Channel>()?.let { clazz to it }
+                    clazz.findAnnotation<Channel>()?.let { clazz to it },
+                    clazz.findAnnotation<AsyncApiComponent>()?.let { clazz to it}
                 )
             }
             .mapNotNull { (clazz, annotation) ->
                 when (annotation) {
                     is Message -> messageProcessor.process(annotation, clazz)
                     is Schema -> schemaProcessor.process(annotation, clazz)
                     is Channel -> channelProcessor.process(annotation, clazz).also {
-                        componentToChannelMapping[clazz.java.simpleName] =
-                            annotation.value.takeIf { it.isNotEmpty() } ?: clazz.java.simpleName
+                        if (clazz.annotations.any { it is Channel }) {
+                            componentToChannelMapping[clazz.java.simpleName] =
+                                annotation.value.takeIf { it.isNotEmpty() } ?: clazz.java.simpleName
+                        }
+                    }
+                    is AsyncApiComponent -> asyncApiComponentProcessor.process(annotation, clazz).also { processedComponents ->
+                        processedComponents.channels?.forEach { (channelName, _) ->
+                            componentToChannelMapping[channelName] = channelName
```

Comment:
You could iterate over the class functions here already. Then you can change the component processor to process the functions instead of the class: `private val asyncApiComponentProcessor: AnnotationProcessor<AsyncApiComponent, KFunction<*>>`
Also you need to use `componentToChannelMapping[function.name] = annotation.value` because there could be multiple channels in the class.